### ISSUE TITLE
Why attention mask is -10000 but not * 0?

### ISSUE THREAD

**Issue by chenxran** [huggingface/transformers#12109]
I am reading the code in Roberta and I found that the implementation of avoiding padding token to be calculated in self-attention is to minus its value by 10000 in function `get_extended_attention_mask`.

 I am wondering that why implement the mask by directly multiplying the value of padding tokens with zero?

**Comment by NielsRogge** [comment]
Hi,

searching previous Github issues, I found this one, which might help you: #542

**Comment by chenxran** [comment]
Thanks for the reference! While I consider that maybe implementing a softmax function that allows masking after e^x may also be an approach to implement `attention_mask`?
